R/print.bayesfactor_models.R

```diff
@@ -1,21 +1,23 @@
 #' @export
 print.bayesfactor_models <- function(x, digits = 2, log = FALSE, ...) {
   BFE <- x
-  denominator <- attr(BFE,'denominator')
-  grid.type <- attr(BFE,'BF_method')
+  denominator <- attr(BFE, 'denominator')
+  grid.type <- attr(BFE, 'BF_method')
 
   BFE_ <- as.data.frame(BFE)
   if (!log) BFE_$log.BF <- exp(BFE_$log.BF)
 
-  rownames(BFE_) <- paste0('[',seq_len(nrow(BFE_)),'] ',BFE_$Model, '   ')
+  # indicate null-model
+  BFE_$Model[BFE_$Model=="1"] <- "(Intercept only)"
+
+  rownames(BFE_) <- paste0('[', seq_len(nrow(BFE_)), '] ', BFE_$Model, '   ')
   denM <- rownames(BFE_)[denominator]
-  BFE_ <- BFE_[-denominator,'log.BF',drop = FALSE]
+  BFE_ <- BFE_[-denominator, 'log.BF', drop = FALSE]
   colnames(BFE_) <- ""
 
   cat('Bayes factor analysis\n--------------')
-  print.data.frame(BFE_,digits = digits)
+  print.data.frame(BFE_, digits = digits)
   cat('\nAgainst denominator:\n\t', denM)
-  cat('\n---\nBayes factor type: ', grid.type)
-  if (log) cat('\nPresenting log(BF)')
-  invisible(x)
```

Comment:
@strengejacke Do we not want to return the original object invisibly?

Reply:
I don't think this is necessary. If you need the object, you call `x <- bayesfactor_models()`. If the print-method returns an object, you can call `x <- print(bayesfactor_models())` to save the object, so there's no benefit.

Reply:
Edge case: using print as a side-effect:

````
library(bayestestR)
library(magrittr)

lm1 <- lm(Sepal.Length ~ 1, data = iris)
lm2 <- lm(Sepal.Length ~ Species, data = iris)
lm3 <- lm(Sepal.Length ~ Species + Petal.Length, data = iris)
lm4 <- lm(Sepal.Length ~ Species * Petal.Length, data = iris)

bayesfactor_models(lm1, lm2, lm3, lm4, denominator = 1) %>% 
  print() %>%
  bayesfactor_inclusion()
````

Reply:
You can do this like:

```r
bayesfactor_models(lm1, lm2, lm3, lm4, denominator = 1) %T>% 
  print() %>%
  bayesfactor_inclusion()
```

Reply:
What if I only loaded `dplyr`? :P

Reply:
It seems to be standard practice in R for printing methods to return the input object (e.g. `stats`'s `print` functions). I'm adding this back in.